### ISSUE TITLE
Training using gt_box with sgcls and predcls

### ISSUE THREAD

**Issue by bknyaz** [rowanz/neural-motifs#93]
Hi,

Thanks for the great code. I have two questions related to #61.

**1.** In the paper you are saying that "We train MOTIFNET on **ground truth boxes**, with the objective to predict object labels and to predict edge labels given ground truth object labels."
So in the scripts you provide here: https://github.com/rowanz/neural-motifs/blob/master/scripts/train_models_sgcls.sh
there should be flag `-gt_box` ?

**2.** Also, the Predicate Classification results in your paper are reported for the model trained using `-m sgcls`, right? Or you trained with the `-m predcls`, i.e. only optimizing the predicate loss (rel_loss)?

**Comment by rowanz** [comment]
1. Sorry for the confusion, that flag doesn't actually do anything, instead you need to specify the evaluation mode like `-m sgcls` or `-m predcls`. 

2. Yep, training in `sgcls` mode optimizes the exact same objective as `predcls`, just we add an extra loss on classifying the objects, so models trained in that mode can be evaluated in both settings (that's what we did). The only thing that's different is `sgdet` mode, which assumes a different data distribution where the GT boxes aren't provided.

**Comment by bknyaz** [comment]
Thanks @rowanz for your reply! I don't know why I got confused regarding 1 :)
I also assume lr must be 0.001 as in the scripts, not 0.006 as mentioned in the paper (Section 5.2)?

**Comment by rowanz** [comment]
Actually, I believe the learning rate in the paper `0.006` should be correct - my bad! I must have tried a few values of the hyperparameter and left the wrong value in the scripts by accident. However, I also recall that the exact hyperparameters didn't matter that much.

**Comment by bknyaz** [comment]
Yeah, I just reproduced results with your scripts (i.e. lr 0.001), so it's probably not that important indeed. Thanks!

**Comment by bknyaz** [comment]
Hi, just noticed that in https://github.com/rowanz/neural-motifs/blob/master/models/train_rels.py#L192 you multiply learning rate by batch size and the number of gpus, so since I run with default argument lr=0.001 and -b 6 and single gpu, it means that the actual learning rate was 0.006 as mentioned in the paper. So, it's good.
